### PR TITLE
[WIP] Adding log stream endpoint with websockets

### PR DESCRIPTION
**Pull request by kcmartin**
Added the beginnings of the stream log endpoint, can you help with this? Thanks!
@sepulworld @seeingidog2

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -223,6 +223,39 @@ def _get_all_pods():
         pods.append(i.metadata.name)
     return pods
 
+# work in progress for log stream endpoint
+@api.route("/stream_logs")
+def stream_logs_html(req, resp):
+    """Route to stream logs"""
+    resp.content = api.template(
+        'stream_logs.html',
+        log_stream=_get_log_stream()
```
Comment (seeingidog2) [inline api.py]: You'll want to pass in a few parameters here, name and namespace at least

File: api.py [hunk 2]
```diff
@@ -223,6 +223,39 @@ def _get_all_pods():
...
+    pretty = 'pretty_example' # str | If 'true', then the output is pretty printed. (optional)
+    previous = true # bool | Return previous terminated container logs. Defaults to false. (optional)
+    since_seconds = 56 # int | A relative time in seconds before the current time from which to show logs. If this value precedes the time a pod was started, only logs since the pod start will be returned. If this value is in the future, no logs will be returned. Only one of sinceSeconds or sinceTime may be specified. (optional)
+    tail_lines = 56 # int | If set, the number of lines from the end of the logs to show. If not specified, logs are shown from the creation of the container or sinceSeconds or sinceTime (optional)
+    timestamps = true # bool | If true, add an RFC3339 or RFC3339Nano timestamp at the beginning of every line of log output. Defaults to false. (optional)
+
+    logs = []
+    ret = k8s_client.read_namespaced_pod_log(name, namespace, container=container, follow=follow, limit_bytes=limit_bytes, pretty=pretty, previous=previous, since_seconds=since_seconds, tail_lines=tail_lines, timestamps=timestamps)
+    # do I need this for loop ??
```
Comment (seeingidog2) [inline api.py]: No you don't. The docs show this function returns a string.

File: api.py [hunk 3]
```diff
@@ -223,6 +223,39 @@ def _get_all_pods():
...
+    resp.content = api.template(
+        'stream_logs.html',
+        log_stream=_get_log_stream()
+    )
+
+def _get_log_stream():
+
+# Not sure where this block should go, and what items we actually need
+    # Do I need this line or is equivalent to 'k8s_client.read_namespaced_pod_log'?
+    api_instance = kubernetes.client.CoreV1Api(kubernetes.client.ApiClient(configuration))
```
Comment (seeingidog2) [inline api.py]: You don't need this. There's already a kubernetes client instantiated earlier. Use that.

File: api.py (hunk 1 above)
Comment (kcmartin) [reply api.py]: I added a commit.

**Comment by kcmartin** [comment]
Passes Travis build now, but I get an "internal server error" on my endpoint; will continue to troubleshoot.

**Comment by sepulworld** [comment]
This required quite a bit of refactoring (using k8s asyncio supported library, javascript functions, etc) Did most of work here: https://github.com/silvermullet/dinghy-ping/pull/89 if you want to help review and test that would rock! probably close this out in favor of PR 89 for now

**Comment by kcmartin** [comment]
Closing this out in favor of #89

**Pull request closed without merging**